### ISSUE TITLE
sqlite3_native.bundle: [BUG] Segmentation fault

### ISSUE THREAD

**Issue by jimhj** [ricardochimal/taps#114]
when I run heroku db:pull to export PostgreSql database to locally mysql database, i got the following error:

```
/Users/huangjin/.rvm/gems/ruby-1.9.3-p125/gems/sqlite3-1.3.6/lib/sqlite3/sqlite3_native.bundle: [BUG] Segmentation fault
ruby 1.8.7 (2010-01-10 patchlevel 249) [universal-darwin11.0]

[1]    2156 abort      heroku db:pull
```

is that a bug?

**Comment by goopie** [comment]
I just started having this issue was well. It started after I updated the heroku binaries.

-- edit - This is the error I receive.

/Users/pauld/.rvm/gems/ruby-1.9.2-p290@greenp/gems/sqlite3-1.3.6/lib/sqlite3/sqlite3_native.bundle: [BUG] Segmentation fault
ruby 1.8.7 (2010-01-10 patchlevel 249) [universal-darwin10.0]

Abort trap

**Comment by jimhj** [comment]
@goopie  I just gave up to export database from Heroku after wasted couple days trying to solve this problem.

**Comment by goopie** [comment]
i went around it by reverting my version of ruby back to 1.8.7 using RVM... a pain, but that did get it to work.

**Comment by subblue** [comment]
See this post: 
http://stackoverflow.com/questions/9389832/heroku-dbpush-fails-with-segfault
reinstalling the Heroku gem fixed it for me.

**Comment by LucasLMartini** [comment]
I believe this is a bug related to the Heroku _toolbelt_ installation in Mac OS X specifically. Are you all on OS X? This probably has nothing to do with Taps

**Comment by constantm** [comment]
Had the same issue after installed Mountain Lion. Got it working by completely uninstalling Heroku gem and Heroku Toolbelt, then reinstalling it from scratch:

```
sudo rm -rf /usr/local/heroku 
sudo rm -rf /usr/bin/heroku
gem uninstall heroku
```

I then reinstalled the latest version of Heroku toolbelt after which I installed the Heroku gem again.

```
gem install heroku
```

Think it's safe to say this doesn't directly involve Taps - it looks like Heroku is trying to use Ruby 1.8 with 1.9 gems?

I'm using rvm 1.15.8 with Ruby 1.9.2-p320 installed and set as default. After reinstalling, the Heroku gem in use is heroku-gem/2.31.5 (x86_64-darwin12.1.0) ruby/1.9.2

**Comment by ZachBeta** [comment]
After the deprecation warnings, I decided to 

```
gem uninstall heroku
```

and use the toolbelt for OSX instead, and ran into this same error.  

For some reason the binary isn't following my rvm ruby and is picking up the OSX default version of ruby.  

**TLDR**: I had to reinstall the heroku gem to get it working again

```
gem install heroku
```

**Comment by will** [comment]
heroku db:push and pull will soon be replaced with pg:push and pull, which won't use taps, so it wouldn't have this problem. It'll have all of its own, new problems! If you want a preview check out https://github.com/heroku/heroku-pg-extras/issues/42
